internal/service/s3/object.go

```diff
@@ -306,12 +313,6 @@ func resourceObjectRead(ctx context.Context, d *schema.ResourceData, meta interf
 		return sdkdiag.AppendFromErr(diags, err)
 	}
 
-	if tags, err := objectListTags(ctx, conn, bucket, key, optFns...); err == nil {
-		setTagsOut(ctx, Tags(tags))
-	} else if !tfawserr.ErrHTTPStatusCodeEquals(err, http.StatusNotImplemented) { // Directory buckets return HTTP status code 501, NotImplemented.
-		return sdkdiag.AppendErrorf(diags, "listing tags for S3 Bucket (%s) Object (%s): %s", bucket, key, err)
-	}
```

Comment:
Will need to test with objects in directory buckets and `default_tags` configured.

Reply:
I think that should be covered by `TestAccS3Object_DirectoryBucket_DefaultTags_providerOnly` (https://github.com/hashicorp/terraform-provider-aws/blob/be5c51b1701cfb845364e3ca60dab2274089e698/internal/service/s3/object_test.go#L1800)

Reply:
Huh, the test no longer passes